Integrations/python/deephaven/learn/__init__.py

```diff
@@ -0,0 +1,190 @@
+
+
+def _defineSymbols():
+    """
+    Defines appropriate java symbol, which requires that the jvm has been initialized through the :class:`jpy` module,
+    for use throughout the module AT RUNTIME. This is versus static definition upon first import, which would lead to an
+    exception if the jvm wasn't initialized BEFORE importing the module.
+
+    Raises:
+        SystemError : if no JVM has been initialized.
+    """
+
+    if not jpy.has_jvm():
+        raise SystemError("No java functionality can be used until the JVM has been initialized through the jpy module")
+
+    global _Input_, _Output_, _Computer_, _Scatterer_
+    if _Input_ is None:
+        _Input_ = jpy.get_type("io.deephaven.integrations.learn.Input")
+        _Output_ = jpy.get_type("io.deephaven.integrations.learn.Output")
+        _Computer_ = jpy.get_type("io.deephaven.integrations.learn.Computer")
+        _Scatterer_ = jpy.get_type("io.deephaven.integrations.learn.Scatterer")
+
+
+# every module method that invokes Java classes should be decorated with @_passThrough
+@wrapt.decorator
+def _passThrough(wrapped, instance, args, kwargs):
+    """
+    For decoration of module methods, to define necessary symbols at runtime
+
+    Args:
+        wrapped  : the method to be decorated
+        instance : the object to which the wrapped function was bound when it was called
+        args     : the argument list for `wrapped`
+        kwargs   : the keyword argument dictionary for `wrapped`
+
+    Returns:
+        the decorated version of the method.
+    """
+    _defineSymbols()
+    return wrapped(*args, **kwargs)
+
+
+@_passThrough
+class Input:
+    """
+    Input specifies how to gather data from a Deephaven table into a Python object.
+    """
+    def __init__(self, colNames, gatherFunc):
+        """
+        Args:
+            colNames   : column name or list of column names from which to gather input.
+            gatherFunc : python function that determines how input gets transformed into a python object.
+        """
+        self.input = _Input_(colNames, gatherFunc)
+
+    def __str__(self):
+        return "Columns: " + str([self.input.getColNames()[i] for i in range(len(self.input.getColNames()))])
+
+
+@_passThrough
+class Output:
+    """
+    Output specifies how to scatter data from a Python object into a table column.
+    """
+    def __init__(self, colName, scatterFunc, type=None):
+        """
+        Args:
+            colName         : name of the new column that will store results.
+            scatterFunc     : python function that determines how data is taken from a python object into a Deephaven table.
+            type (optional) : desired data type of the new output column. Must be one of int, long, double, float, boolean, char, String.
+        """
+        self.output = _Output_(colName, scatterFunc, type)
+
+    def __str__(self):
+        print("Column: " + self.output.getColName())
+        print("Type: " + self.output.getType())
```

Comment:
1) I would implement the string in the java class as a `toString()` method.  Then call the method here.  Then either language can print out the details.

2) The contract for `__str__` seems to indicate that it should return a string instead of printing stuff.